### DISCUSSION TITLE
How does SumatraPDF know that GhostScript is installed?

### DISCUSSION THREAD

**Discussion by hfpop** [sumatrapdfreader/sumatrapdf#4008] [Q&A]
Hello,

I am on Windows, and this is a Windows 64bit question. SumatraPDF uses GhostScript when it is installed. But how does it know gs is installed? My setup is a manually created folder with the gs dll there, with no actual install of gs. From what I tried, if it finds gsdll64.dll somewhere in the path, it’s ok. I also tried with system variables, but I had no succes here. What exactly does sumatra check here?

Thank you.

**Comment by GitHubRulesOK** [comment]
it looks for other files primarily in the registry so an installed xps viewer acrobat and several other installed external viewers and also Ghostscript registry entry and also find some via path. see this area of code
https://github.com/sumatrapdfreader/sumatrapdf/blob/d4c44b2587cfb17476c8b243aeee09b1acf90569/src/EnginePs.cpp#L49

https://github.com/sumatrapdfreader/sumatrapdf/blob/d4c44b2587cfb17476c8b243aeee09b1acf90569/src/EnginePs.cpp#L81

**Comment by hfpop** [comment]
Thank you indeed!
